File: app/src/main/java/br/com/alexandre/friday/recognize/chain/StreetChain.java
package br.com.alexandre.friday.recognize.chain;

import java.util.Arrays;
import java.util.Collection;

import br.com.alexandre.friday.recognize.RecognizerChain;
import br.com.alexandre.friday.recognize.RecognizerRequest;

public class StreetChain extends RecognizerChain {
    public Collection<String> execute(RecognizerRequest request) {
        if (request.getSubThoroughfare() == null || request.getSubThoroughfare().trim().isEmpty()) {
            return Arrays.asList(String.format("Você está na %s", request.getThoroughfare()));
        } else {
            return Arrays.asList(String.format("Você está na %s, número %s", request.getThoroughfare(), request.getSubThoroughfare()));
        }
    }

    public boolean apply(RecognizerRequest request) {
        return  (request.getQ().contains("qual") &&
                request.getQ().contains("é") &&
                request.getQ().contains("essa") &&
                request.getQ().contains("rua"))
                ||
                (request.getQ().contains("que") &&
                        request.getQ().contains("rua") &&
                        request.getQ().contains("é") &&
                        request.getQ().contains("essa"));
    }
}

File: recognition/src/main/java/br/com/alexandre/friday/recognize/chain/StreetChain.kt
package br.com.alexandre.friday.recognize.chain

import br.com.alexandre.friday.recognize.RecognizerChain
import br.com.alexandre.friday.recognize.RecognizerRequest
import java.util.*

class StreetChain: RecognizerChain() {
    override fun execute(request: RecognizerRequest?): MutableCollection<String> {
        var response = if (request?.subThoroughfare == null || request.subThoroughfare.trim { it <= ' ' }.isEmpty()) {
            String.format("Você está na %s", request?.thoroughfare)
        } else {
            String.format("Você está na %s, número %s", request.thoroughfare, request.subThoroughfare)
        }
        return Arrays.asList(response)
    }

    override fun apply(request: RecognizerRequest?): Boolean {
        return request?.q!!.contains("qual") &&
                request.q!!.contains("é") &&
                request.q!!.contains("essa") &&
                request.q!!.contains("rua")
                ||
                request.q!!.contains("que") &&
                request.q!!.contains("rua") &&
                request.q!!.contains("é") &&
                request.q!!.contains("essa")
    }
}